elide-model-config/src/main/java/com/yahoo/elide/modelconfig/validator/DynamicConfigValidator.java

```diff
@@ -174,57 +175,114 @@ private static void validateInheritance(ElideTableConfig tables, Table table, Se
+                                }
+                            });
+                            return result;
+                        }, false
+                );
                 table.setJoins(new ArrayList<Join>(joins.values()));
-            }
-        }
-    }
 
-    private Map<String, Measure> getInheritedMeasures(Table table, Map<String, Measure> measures) {
-        table.getMeasures().forEach(m -> {
-            if (!measures.containsKey(m.getName())) {
-                measures.put(m.getName(), m);
+                String schema = (String) getInheritedAttribute(table, null,
+                        (tab, result) -> {
+                            return tab.getSchema();
+                        }, true
+                );
+                table.setSchema(schema);
+
+                String dbConnectionName = (String) getInheritedAttribute(table, null,
+                        (tab, result) -> {
+                            return tab.getDbConnectionName();
+                        }, true
+                );
+                table.setDbConnectionName(dbConnectionName);
+
+                String readAccess = (String) getInheritedAttribute(table, null,
+                        (tab, result) -> {
+                            return tab.getReadAccess();
+                        }, true
+                );
+                table.setReadAccess(readAccess);
+
+                String sql = (String) getInheritedAttribute(table, null,
+                        (tab, result) -> {
+                            return tab.getSql();
+                        }, true
+                );
+                table.setSql(sql);
+
+                String tableName = (String) getInheritedAttribute(table, null,
+                        (tab, result) -> {
+                            return tab.getTable();
+                        }, true
+                );
+                table.setTable(tableName);
             }
-        });
-        if (table.hasParent()) {
-            getInheritedMeasures(table.getParent(this.elideTableConfig), measures);
         }
-        return measures;
     }
 
-    private Map<String, Dimension> getInheritedDimensions(Table table, Map<String, Dimension> dimensions) {
-        table.getDimensions().forEach(dim -> {
-            if (!dimensions.containsKey(dim.getName())) {
-                dimensions.put(dim.getName(), dim);
-            }
-        });
-        if (table.hasParent()) {
-            getInheritedDimensions(table.getParent(this.elideTableConfig), dimensions);
-        }
-        return dimensions;
-    }
 
-    private Map<String, Join> getInheritedJoins(Table table, Map<String, Join> joins) {
-        table.getJoins().forEach(join -> {
-            if (!joins.containsKey(join.getName())) {
-                joins.put(join.getName(), join);
-            }
-        });
+    private Object getInheritedAttribute(Table table, Object result, Inheritance action, boolean recurseOnlyIfNull) {
```

Comment:
The last parameter recurseOnlyIfNull is a tell that this really ought to be two separate functions.